PermanentOrg/web-app
# Update navigation bar for GLAM onboarding

This pull request features a style adjustment to GLAM onboarding. Specifically, the header or navigation bar at the top of the page now matches the one seen in the GLAM designs. The original header is preserved for the current onboarding flow while GLAM is not enabled.

In addition, this PR features some small refactoring work. Right now it mainly involves just moving some components around the filesystem. The Onboarding Module's components folder has a "/glam" subdirectory that I created to hold all of the new "GLAM" components so it was easy to tell which components were new and which ones were old. There has been some inconsistency in whether or not new components land in this folder. I have moved each GLAM component over to the `glam` directory so that it's easy to identify the new and old subcomponents. Ideally there'd also be some kind of "classic" or "legacy" folder for the old component, but since the GLAM components are currently used in the create-new-archive component, it felt misleading to fully separate them. That will probably come in a future refactoring. Before doing that I wanted to improve the onboarding testing workflow a bit, which brings me to...

The other big addition I tacked onto this PR is the addition of **a full "Onboarding Demo" suite of stories in Storybook!** This allows us to test the onboarding flow without having to create a million accounts each time, or even be connected to the API at all. It also allows for fast switching between the various onboarding flows (invitations, glam, etc). If you recall when onboarding was first developed, we had a debugger component to help test with things while it was still under construction.

<details>
<summary>Finally after a few years that debugging functionality lives on in Storybook!</summary>

![onboarding-debug-component](https://github.com/user-attachments/assets/f5da090c-ffb1-48a7-92cc-3d1dd17f436d)

</details>

Steps to test:
1. Create a new account and activate GLAM onboarding (`localStorage.setItem('isGlam', 'true');`)
2. Make sure the header looks good and works

You can also view the header in the new Onboarding Demo on Storybook, though the "sign out" button doesn't fully work in Storybook since there are no user sessions and you cannot be redirected to the sign in screen.
